flink-staging/flink-scala-shell/src/main/scala/org/apache/flink/api/scala/FlinkShell.scala

```diff
@@ -0,0 +1,148 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.scala
+
+
+import java.io.{StringWriter, BufferedReader}
+
+import scala.tools.nsc.Settings
+
+import org.apache.flink.configuration.Configuration
+import org.apache.flink.runtime.minicluster.LocalFlinkMiniCluster
+
+import scala.tools.nsc.interpreter._
+
+
+object FlinkShell {
+
+  val LOCAL = 0;
+  val REMOTE = 1;
+  val UNDEFINED = -1;
+
+  var bufferedReader: BufferedReader = null;
+
+  def main(args: Array[String]) {
+
+    // scopt, command line arguments
+    case class Config(
+        port: Int = -1,
+        host: String = "none",
+        externalJars: Option[Array[String]] = None,
+        flinkShellExecutionMode : Int = UNDEFINED)
+
+    val parser = new scopt.OptionParser[Config]("start-scala-shell.sh") {
+      head ("Flink Scala Shell")
+
+      cmd("local") action {
+        (_,c) => c.copy( host = "none", port = -1, flinkShellExecutionMode = LOCAL)
+      } text("starts Flink scala shell with a local Flink cluster\n") children(
+        opt[(String)] ("addclasspath") abbr("a") valueName("<path/to/jar>") action {
+          case (x,c) =>
+            val xArray = x.split(":")
+            c.copy(externalJars = Option(xArray))
+          } text("specifies additional jars to be used in Flink\n")
+        )
+
+      cmd("remote") action { (_, c) =>
+        c.copy(flinkShellExecutionMode = REMOTE)
+      } text("starts Flink scala shell connecting to a remote cluster\n") children(
+        arg[String]("<host>") action { (h, c) =>
+          c.copy(host = h) }
+          text("remote host name as string"),
+        arg[Int]("<port>") action { (p, c) =>
+          c.copy(port = p) }
+          text("remote port as integer\n"),
+        opt[(String)] ("addclasspath") abbr("a")  valueName("<path/to/jar>") action {
+          case (x,c) =>
+            val xArray = x.split(":")
+            c.copy(externalJars = Option(xArray))
+          } text("specifies additional jars to be used in Flink")
+
+      )
+      help("help") abbr("h") text("prints this usage text\n")
+    }
+
+
+    // parse arguments
+    parser.parse (args, Config () ) match {
+      case Some(config) =>
+        startShell(config.host,
+          config.port,
+          config.flinkShellExecutionMode,
+          config.externalJars)
+
+      case _ => println("Could not parse program arguments")
+    }
+  }
+
+
+  def startShell(
+      userHost : String, 
+      userPort : Int,
+      executionMode : Int,
+      externalJars : Option[Array[String]] = None): Unit ={
+    
+    println("Starting Flink Shell:")
+
+    var cluster: LocalFlinkMiniCluster = null
```

Comment:
Can we make `cluster` a `val` of type `Option[LocalFlinkMiniCluster]` whose value is the value of the `if-else` expression? That way we avoid `null` values which are evil.